BJS2-68810

tanufa25 opened the pull request:
Task with spell events is done

Commit:
BJS2-68810
Task with spell events is done

20cognosce commented on src/main/java/school/faang/bjs2_68810/catch_event/SpellEvent.java:
```diff
@@ -0,0 +1,16 @@
+@ToString
+@Getter
+public class SpellEvent {
+    private int id;
+    private String eventType;
+    private String action;
```
final

20cognosce commented on src/main/java/school/faang/bjs2_68810/catch_event/SpellEvent.java:
```diff
@@ -0,0 +1,16 @@
+import lombok.ToString;
+
+@AllArgsConstructor
+@EqualsAndHashCode
```
У тебя id есть, какой смысл?

20cognosce commented on src/main/java/school/faang/bjs2_68810/catch_event/Main.java:
```diff
@@ -0,0 +1,76 @@
+        hogwartsSpells.printAllSpellEvents();
+
+        // Testing getSpellEventById
+        // ok
```
не писать бессмысленные комментарии

20cognosce commented on src/main/java/school/faang/bjs2_68810/catch_event/HogwartsSpells.java:
```diff
@@ -0,0 +1,57 @@
+@Slf4j
+public class HogwartsSpells {
+    private Map<Integer, SpellEvent> spellByIdMap = new HashMap<>();
+    private Map<String, List<SpellEvent>> spellsByTypeMap = new HashMap<>();
```
final

20cognosce commented on src/main/java/school/faang/bjs2_68810/catch_event/SpellEvent.java:
```diff
@@ -0,0 +1,16 @@
+@Getter
+public class SpellEvent {
+    private int id;
+    private String eventType;
```
enum

20cognosce commented on src/main/java/school/faang/bjs2_68810/catch_event/HogwartsSpells.java:
```diff
@@ -0,0 +1,57 @@
+public class HogwartsSpells {
+    private Map<Integer, SpellEvent> spellByIdMap = new HashMap<>();
+    private Map<String, List<SpellEvent>> spellsByTypeMap = new HashMap<>();
+    private int currentId = 0;
```
Лучше в отдельный класс вынести, который будет чисто айдишники выдавать
сделать его AtomicInteger и в статическом методе делать инкремент и отдавать новый айди

либо использовать UUID.random

В ответственность HogwartsSpells не входит отслеживание и выдача айдишников

20cognosce commented on src/main/java/school/faang/bjs2_68810/catch_event/HogwartsSpells.java:
```diff
@@ -0,0 +1,57 @@
+    private int currentId = 0;
+
+    public void addSpellEvent(String eventType, String actionDescription) {
+        currentId++;
```
.

20cognosce commented on src/main/java/school/faang/bjs2_68810/catch_event/HogwartsSpells.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    public void deleteSpellEvent(int id) {
+        SpellEvent spellEvent = getSpellEventById(id);
```
какой в этом смысл

tanufa25 replied:
Просто в "getSpellEventById(id);" уже расписана логика для выбрасывания исключения, если не найдено событие. Чтобы не дублировать этот код, я сначала вызываю "getSpellEventById(id);" , а потом уже реализовываю логику удаления, учитывая, что null мне уже точно не придёт.

tanufa25 replied:
Или наоборот надо учитывать , что get может поменяться, а функция удаления должна быть автономной?

20cognosce commented on src/main/java/school/faang/bjs2_68810/catch_event/HogwartsSpells.java:
```diff
@@ -0,0 +1,57 @@
+    public void deleteSpellEvent(int id) {
+        SpellEvent spellEvent = getSpellEventById(id);
+
+        spellByIdMap.remove(id);
```
если тут remove итак возвращает значение

Commit:
BJS2-68810
final is added for variables.
id is generating by UUID.randomUUID().
EventType enum is added.
Comments were removed.
getSpellEventById(id) method is removed from deleteSpellEvent logic.
Constructor of SpellEvent is modified.

20cognosce commented on src/main/java/school/faang/bjs2_68810/catch_event/HogwartsSpells.java:
```diff
@@ -0,0 +1,57 @@
+        List<SpellEvent> spellEventList = spellsByTypeMap.get(spellEventType);
+        if (spellEventList == null) {
+            log.error("SpellEvents with type {} are not found", spellEventType);
+            throw new SpellEventNotFoundException("Exception: SpellEvents with type "
+                    + spellEventType + " are not found");
```
String.format

20cognosce commented on src/main/java/school/faang/bjs2_68810/catch_event/HogwartsSpells.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    public void printAllSpellEvents() {
+        for (Map.Entry<UUID, SpellEvent> entry : spellByIdMap.entrySet()) {
```
forEach
